RUFAS/input_manager.py

```diff
@@ -91,3 +91,113 @@ def _load_data(self) -> None:
                                    info_map)
             except Exception as e:
                 raise e
+
+    def _validate_data(self, eager_termination: bool = True) -> bool:
+        """
+        Validates input data and attempts to fix any invalid input data.
+
+        Parameters
+        ----------
+        eager_termination : bool, default=True
+            If true, the process will be terminated upon finding invalid data.
+
+        Returns
+        -------
+        bool
+            True if all data is valid; False otherwise.
+        """
+        info_map = {"class": self.__class__.__name__,
+                    "function": self._validate_data.__name__,
+                    }
+        valid_elements_counter = 0
+        invalid_elements_counter = 0
+        fixed_elements_counter = 0
+        unfixed_elements_counter = 0
+        invalid_critical_elements_counter = 0
+        total_elements_checked_counter = 0
+
+        for key in self.__pool.keys():
+            for variable, value in self.__pool[key].items():
+                total_elements_checked_counter += 1
+                property_map_key = self.__metadata["files"][key]["properties"]
+                if self._validate_element(variable, value):
+                    valid_elements_counter += 1
+                else:
+                    invalid_elements_counter += 1
+                    is_data_critical = "default" not in self.__metadata["properties"][property_map_key][variable].keys()
+                    if is_data_critical:
+                        invalid_critical_elements_counter += 1
+                    is_data_fixable = self._fix_data(variable, value)
+                    if is_data_fixable:
+                        fixed_elements_counter += 1
+                    elif not is_data_fixable and eager_termination:
+                        return False
+                    else:
+                        unfixed_elements_counter += 1
+
+        om.add_log("Total Valid Elements", f"{valid_elements_counter} valid elements found.", info_map)
```

Comment:
Simplify:
```suggestion
        om.add_log("Total Valid Elements", f"{valid_elements_counter=}", info_map)
```
The same applies to the rest.